crates/re_log_types/src/size_bytes.rs

```diff
@@ -349,14 +352,25 @@ fn estimated_bytes_size(array: &dyn Array) -> usize {
                 }
 
                 let fields = array.fields();
-                let values_start = offsets[0] as usize;
-                let values_end = offsets[offsets.len() - 1] as usize;
-                array
-                    .types()
-                    .iter()
-                    .copied()
-                    .map(|ty| {
-                        fields.get(ty as usize).map_or(0, |x| {
+                let types: IntSet<_> = array.types().iter().copied().collect();
+                types
+                    .into_iter()
+                    .map(|cur_ty| {
+                        let mut indices = array
+                            .types()
+                            .iter()
+                            .enumerate()
+                            .filter_map(|(idx, &ty)| (ty == cur_ty).then_some(idx));
+
+                        let idx_start = indices.next().unwrap_or_default();
+                        let mut idx_end = idx_start;
+                        for idx in indices {
+                            idx_end = idx;
+                        }
```

Comment:
```suggestion
                        let idx_end = indices.last().unwrap_or_default();
```

Reply:
Yeah... I didn't go there because it very much looks like this some magic `0(1)` function, which feels like I'm betraying the reader...